server/task_scheduler.go

```diff
@@ -122,13 +146,17 @@ func (s *TaskScheduler) Run(ctx context.Context, wg *sync.WaitGroup) {
 						)
 						continue
 					}
-
-					if _, ok := s.runningExecutors[task.ID]; ok {
-						continue
-					}
+					s.mu.Lock()
 					s.runningExecutors[task.ID] = executorGetter()
+					s.mu.Unlock()
 
 					go func(task *api.Task, executor TaskExecutor) {
+						defer func() {
+							s.mu.Lock()
+							delete(s.runningExecutors, task.ID)
+							s.taskProgress.Delete(task.ID)
```

Comment:
The way it writes makes it unclear what does mu protect.

Does it only protect runningExecutors as well as taskProgress?

Reply:
Done.